modules/thermal_hydraulics/doc/content/source/controllogic/SmootherControl.md

```diff
@@ -0,0 +1,21 @@
+# SmootherControl
+
+The `SmootherControl` is a controller that smooths the input signal using a moving average technique.
+
+It receives an [!param](/ControlLogic/SmootherControl/input) value  and stores it into a vector of size [!param](/ControlLogic/SmootherControl/n_points). When the vector reaches its maximum capacity (`n_points`), the oldest value (first element) is removed, and the current input is added to the end of the vector.
+Then it calculates the average of the values stored in the vector. The value of the `output` data is computed as:
+
+!equation
+output = \frac{1}{s(v)} \Sigma^{s(v)}_{i=1} v_i\\  \\
+s(v) \leq N
+
+where $v$ is the vector containing the input values, $s(v)$ is the current size of the vector, and $N$ is the maximum size of the vector, defined by [!param](/ControlLogic/SmootherControl/n_points).
+
+Since, by default, the control logic is executed on the initial condition, the first output will be 0.
```

Comment:
```suggestion
```

Reply:
Not true - the output value correct; it's just that a PP would report it incorrectly.